airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/Batch.kt

```diff
@@ -45,6 +50,8 @@ import com.google.common.collect.TreeRangeSet
  * ```
  */
 interface Batch {
+    val id: String?
```

Comment:
Will this eventually become non optional?

Reply:
No. The id is for the use case when you want to conflate a series of batches. For example, you are uploading a series of parts, and you want the whole group to be marked PERSISTED only when the entire group is done.

If you're processing batches individually, you won't want this.

I think I can make this more clear (and correct) by changing this to `groupId`.